numalogic/registry/redis_registry.py

```diff
@@ -123,20 +124,38 @@ def __get_artifact_data(
             extras={
                 "timestamp": float(artifact_timestamp.decode()),
                 "version": artifact_version.decode(),
+                "source": self._STORETYPE,
             },
         )
 
-    def __load_latest_artifact(self, key: str) -> ArtifactData:
+    def __load_latest_artifact(self, key: str) -> tuple[ArtifactData, bool]:
+        """
+        Load the latest artifact from the registry.
+
+        Args:
+            key: full model key.
+
+        Returns
+        -------
+            ArtifactData and a boolean flag indicating if the artifact was loaded from cache.
+
+        Raises
+        ------
+            ModelKeyNotFound: If the model key is not found in the registry.
+        """
         cached_artifact = self._load_from_cache(key)
         if cached_artifact:
             _LOGGER.debug("Found cached artifact for key: %s", key)
-            return cached_artifact
+            return cached_artifact, True
```

Comment:
can you make the return an explicit tuple?

Reply:
Pylint complains about redundant parenthesis